GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -198,28 +201,43 @@ private bool TryRunCleanUp(out string error)
         {
             error = null;
 
-            string errorMessage = null;
+            string errorMessage = string.Empty;
             if (!this.LaunchInsideSpinner(
                 () =>
                 {
-                    if (!this.TryReleaseUpgradeLock(out errorMessage))
+                    bool success = true;
+                    string unlockError;
+                    if (!this.TryReleaseUpgradeLock(out unlockError))
                     {
-                        return false;
+                        errorMessage += unlockError + Environment.NewLine;
+                        success = false;
                     }
 
-                    if (this.remount && !this.preRunChecker.TryMountAllGVFSRepos(out errorMessage))
+                    string remountError;
+                    if (this.remount && !this.preRunChecker.TryMountAllGVFSRepos(out remountError))
                     {
                         EventMetadata metadata = new EventMetadata();
                         metadata.Add("Upgrade Step", nameof(this.TryRunCleanUp));
-                        this.tracer.RelatedError(metadata, $"{nameof(this.preRunChecker.TryMountAllGVFSRepos)} failed. {errorMessage}");
-                        return false;
+                        this.tracer.RelatedError(metadata, $"{nameof(this.preRunChecker.TryMountAllGVFSRepos)} failed. {remountError}");
+                        errorMessage += remountError + Environment.NewLine;
+                        success = false;
                     }
 
-                    return true;
+                    string downloadsCleanupError;
+                    if (this.deleteDownloadedAssets && !this.upgrader.TryCleanup(out downloadsCleanupError))
+                    {
+                        EventMetadata metadata = new EventMetadata();
+                        metadata.Add("Upgrade Step", nameof(this.TryRunCleanUp));
+                        this.tracer.RelatedError(metadata, $"{nameof(this.upgrader.TryCleanup)} failed. {downloadsCleanupError}");
+                        errorMessage += downloadsCleanupError + Environment.NewLine;
+                        success = false;

```

Comment:
Why do we need to fail the upgrade (in the UX) if this step fails? The upgrade has actually completed successfully right? And running upgrade again will just tell them they're caught up. If that's true, then this is not an error.